packages/backend-tasks/src/tasks/PluginTaskSchedulerJanitor.test.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { TestDatabases } from '@backstage/backend-test-utils';
+import { Knex } from 'knex';
+import { Duration } from 'luxon';
+import waitForExpect from 'wait-for-expect';
+import { migrateBackendTasks } from '../database/migrateBackendTasks';
+import { DbTasksRow, DB_TASKS_TABLE } from '../database/tables';
+import { PluginTaskSchedulerJanitor } from './PluginTaskSchedulerJanitor';
+
+const insertTask = async (knex: Knex, task: DbTasksRow) => {
+  return knex<DbTasksRow>(DB_TASKS_TABLE)
+    .insert(task)
+    .onConflict('id')
+    .merge(['settings_json']);
+};
+
+const getTask = async (knex: Knex): Promise<DbTasksRow> => {
+  return (await knex<DbTasksRow>(DB_TASKS_TABLE))[0];
+};
+
+describe('PluginTaskSchedulerJanitor', () => {
+  const logger = getVoidLogger();
+  const databases = TestDatabases.create({
+    ids: [
+      /* 'MYSQL_8' not supported yet */
```

Comment:
the plugins/catalog-backend/src/database/DefaultProcessingDatabase.test.ts uses the MYSQL_8.
why in this case the mysql 8 not work ?

Reply:
That's the catalog, not the backend-tasks package. Every plugin that has any database commands, needs to separately make sure that their stuff works with the required database engines. Mysql is very optional, and almost nothing works with it; postgres is the only strictly supported engine in production. Feel free to try the tests with that commented-out ID added to see what mysql doesn't like about the migrations in their current state.